builtin/mainmenu/settings/dlg_settings.lua

```diff
@@ -641,11 +641,20 @@ local function buttonhandler(this, fields)
 		local value = core.is_yes(fields.show_advanced)
 		core.settings:set_bool("show_advanced", value)
 		write_settings_early()
+	end
+
+	if fields.enable_touch ~= nil then
```

Comment:
where is the field this code handles defined?

Reply:
I'm afraid I don't understand the question.

Reply:
I can not find it either.

Reply:
The contents of `fields` are set according to the state of the formspec and/or buttons pressed.
If there is no formspec field named "enable_touch" defined then this key can never show up in the table.
I couldn't find one, hence the question.

Reply:
The field is auto-generated from `settingtypes.txt`. That defines the key.

Reply:
This is misleading placement, given that it's next to show_advanced but the actual checkbox is defined as a component based on the setting. I'd recommend adding a comment at least, or maybe moving this code to `make.bool` although maybe that violates separation of responsibiities

Reply:
This is misleading placement, given that it's next to show_advanced but the actual checkbox is defined as a component based on the setting. I'd recommend adding a comment at least, or maybe moving this code to make.bool although maybe that violates separation of responsibiities